monai/transforms/compose.py

```diff
@@ -86,7 +99,13 @@ def __init__(self, transforms=None):
 
     def __call__(self, input_):
         for transform in self.transforms:
-            input_ = transform(input_)
+            # if some transform generated batch list of data in the transform chain,
+            # all the following transforms should apply to every item of the list.
+            if isinstance(input_, list):
+                for i, item in enumerate(input_):
+                    input_[i] = transform(item)
```

Comment:
potentially losing the capability of processing `enumerate(input_)` with multiple workers... personally I don't quite like this idea of treating the image-patch sampling as a composable transform. But I guess I can approve this PR because it doesn't break any existing features

Reply:
Thanks for your review, I already updated code for all comments.